File: backend/src/main/java/com/internbridge/backend/service/LogbookService.java
package com.internbridge.backend.service;

import com.internbridge.backend.dto.request.LogbookRequestDTO;
import com.internbridge.backend.dto.response.LogbookResponseDTO;
import java.util.List;
import java.util.UUID;

public interface LogbookService {
    LogbookResponseDTO createLogbook(LogbookRequestDTO requestDTO);
    LogbookResponseDTO getLogbookById(UUID id);
    List<LogbookResponseDTO> getLogbooksByPlacementId(UUID placementId);
    LogbookResponseDTO updateLogbook(UUID id, LogbookRequestDTO requestDTO);
    LogbookResponseDTO updateCompanyStatus(UUID id, String status);
    LogbookResponseDTO updateLecturerStatus(UUID id, String status);
    void deleteLogbook(UUID id);
}

File: frontend/src/api/logbookService.ts
import { legacyApi } from './axiosConfig';

export interface LogbookRequest {
  placementId: string;
  recordDate: string;
  hoursWorked: number;
  tasksCompleted: string;
}

export interface LogbookDTO {
  id: string;
  placementId: string;
  recordDate: string;
  hoursWorked: number;
  tasksCompleted: string;
  companyStatus: string;
  lecturerStatus: string;
}

const logbookService = {
  createLogbook: async (data: LogbookRequest): Promise<LogbookDTO> => {
    const response = await legacyApi.post<LogbookDTO>('/logbooks', data);
    return response.data;
  },

  getLogbooksByPlacement: async (placementId: string): Promise<LogbookDTO[]> => {
    const response = await legacyApi.get<LogbookDTO[]>(`/logbooks/placement/${placementId}`);
    return response.data;
  },

  getLogbookById: async (id: string): Promise<LogbookDTO> => {
    const response = await legacyApi.get<LogbookDTO>(`/logbooks/${id}`);
    return response.data;
  },

  updateLogbook: async (id: string, data: LogbookRequest): Promise<LogbookDTO> => {
    const response = await legacyApi.put<LogbookDTO>(`/logbooks/${id}`, data);
    return response.data;
  },

  deleteLogbook: async (id: string): Promise<void> => {
    await legacyApi.delete(`/logbooks/${id}`);
  },

  updateCompanyStatus: async (id: string, status: string): Promise<LogbookDTO> => {
    const response = await legacyApi.patch<LogbookDTO>(`/logbooks/${id}/company-status?status=${status}`);
    return response.data;
  },

  updateLecturerStatus: async (id: string, status: string): Promise<LogbookDTO> => {
    const response = await legacyApi.patch<LogbookDTO>(`/logbooks/${id}/lecturer-status?status=${status}`);
    return response.data;
  },
};

export default logbookService;
